### DISCUSSION TITLE
Difficult to Rule Trigger when using MI32Option6 1

### DISCUSSION THREAD

**Discussion by kabongsteve** [arendst/Tasmota#15119] [Q&A] [answered]
I'm trying to use a Bluetooth Window Sensor to lock/unlock Shutter operations (Open window can break roller shutter).
EG
`RULE 2 ON DOOR4b7069#Btn=1 DO ShutterLock 0 ENDON On DOOR4b7069#Btn=0 DO ShutterLock 1 ENDON`
console log:
`17:27:09.117 MQT: tele/shutter1/SENSOR = {"Time":"2022-03-13T17:27:09","DOOR4b7069":{"mac":"e4aaec4b7069","Light":-1,"Btn":2,"RSSI":-68}, ...`

The default "scanning mode" has a delay between when the sensor is trigger and when logged and clok enabled/ disabled.

By using MI32Option6 1, you get more immediate notification from the sensor in the format:
`17:32:12.105 MQT: tele/tasmota_ble/DOOR4b70a7 = {"Time":"2022-03-13T17:32:12","mac":"e4aaec4b70a7","Light":-1,"Btn":-1,"RSSI":-95}`

Unfortunately, the bluetooth also capture messages from other sensors as well (including other dor/window sensors):
`17:29:40.091 MQT: tele/tasmota_ble/DOOR4b70a7 = {"Time":"2022-03-13T17:29:40","mac":"e4aaec4b70a7","Light":-1,"Btn":-1,"RSSI":-95}
17:32:08.088 MQT: tele/tasmota_ble/ATC4fd85c = {"Time":"2022-03-13T17:32:08","mac":"a4c1384fd85c","Temperature":23.2,"Humidity":47.3,"DewPoint":11.3,"Btn":1,"Battery":82,"RSSI":-74,"TempUnit":"C"}
17:32:09.073 MQT: tele/tasmota_ble/ATC9099b2 = {"Time":"2022-03-13T17:32:09","mac":"a4c1389099b2","Temperature":23.3,"Humidity":46.7,"DewPoint":11.2,"Btn":1,"Battery":84,"RSSI":-81,"TempUnit":"C"}
17:32:10.084 MQT: tele/tasmota_ble/DOOR4b3397 = {"Time":"2022-03-13T17:32:10","mac":"e4aaec4b3397","Light":-1,"Btn":2,"RSSI":-66}`

There is NO identifying data in the JSON to state which sensor generated the json, and thus hard to filter in a rule.

My suggestion (if no others), would be to use the same JSON format for both MI32Option6 o or 1, so that the sensor data is nested within the sensor name either way.

EG:
`17:32:12.105 MQT: tele/tasmota_ble/DOOR4b70a7 = {"Time":"2022-03-13T17:27:09","DOOR4b7069":{"mac":"e4aaec4b7069","Light":-1,"Btn":2,"RSSI":-68}`

Thus it wouldn't matter which mode is being used, the rule trigger will always work, and only for the specified sensor.

The same would go for HomeAssistant or other tools using the MQTT log for monitoring window states

**Comment by sfromis** [comment, marked as answer]
If you want a format suitable for rule triggers, you indeed get that by not using `MI32Option6 1`.

For "immediate" notification, I'm using `MI32Option2 1` and get the format well-working for rule triggers.

**Comment by barbudor** [comment]
That could make sense for a `Mi32Option6 2` mode ....
I can have a look tonight

**Comment by barbudor** [comment]
@sfromis 
Using `Mi32Option6 0`+ `Mi32Option2 1`, I get a message from my ATC sensors once every second or so.
Is that what you have and expecting ?

**Reply by barbudor** [reply]
Mi32Option2 for immediate notification is more suitable for a door sensor than Mi32Option6
Would be great to have different behavior for different sensor
Maybe blerry can do that ?
Still haven't got time to dig into it

**Reply by sfromis** [reply]
Well, `MI32Option2 1` reporting obviously depends on the device. And kabongsteve seemed to want "immediate" notification, which implies that waiting till TelePeriod is not preferable.

With a factory-firmware LYWSD03, I see reporting roughly every minute, and I'm fine with that.

**Reply by kabongsteve** [reply]
Thanks all,  Sorry, I'd missed the MI32Option2 setting so will play around with that instead of option6.

Option 6 has the advantage of unique mqtt message per sensor, better for hass operation and configuration

**Reply by sfromis** [reply]
I consider that a disadvantage :-)

AFAIK, HA has no problem processing combined payloads, that already happens extensively for Tasmota wired sensors, where you can have many sensors in the same long JSON payload.

**Comment by barbudor** [comment]
@kabongsteve Here is `Mi32Option6 2` : https://github.com/arendst/Tasmota/pull/15383
Now you can have 1 topic per sensor and still use rules
